Std/Data/List/Lemmas.lean

```diff
@@ -403,6 +403,11 @@ theorem forall_mem_append {p : α → Prop} {l₁ l₂ : List α} :
     (∀ x ∈ l₁ ++ l₂, p x) ↔ (∀ x ∈ l₁, p x) ∧ (∀ x ∈ l₂, p x) := by
   simp only [mem_append, or_imp, forall_and]
 
+theorem forall_mem_iff_forall_index {p : α → Prop} {l : List α} :
+    (∀ x ∈ l, p x) ↔ (∀ i : Fin (l.length), p (l.get i)) :=
+  ⟨fun h i => h _ (get_mem l i.1 i.2),
+  fun h _ hx => by have ⟨_, hi⟩ := mem_iff_get.mp hx; simp only [← hi, h]⟩
+
```

Comment:
```suggestion
theorem forall_mem_iff_forall_index {p : α → Prop} {l : List α} :
    (∀ x ∈ l, p x) ↔ (∀ i : Fin (l.length), p (l.get i)) := by 
  simp [List.mem_iff_get]
  
```

Reply:
What is the use case of this theorem?

Reply:
> What is the use case of this theorem?

Sometimes some propositions of `List` are expressed in indexed form, especially when translating algorithm proof from CS books. 

I struggled with it for hours and it's until this issue then I know the existence of `List.mem_iff_get` which derives this theorem. Without it there may be someone else facing the same trouble again.

Reply:
Sorry about the slow reply here. Let's reverse the `iff`, rename to `forall_get_iff_forall_mem`, and maybe even mark this as `@[simp]`.

Reply:
Done.

Reply:
I think it shouldn't be `@[simp]`; this will not fire reliably anyway (only the RTL direction looks like it would match), and whether it is better to use indexing or membership in a given context is very dependent on other factors.